domains/transfer/Transfer.API/API/Claiming/Api/Controllers/ClaimController.cs

```diff
@@ -0,0 +1,107 @@
+using API.Shared.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Claiming.Api.Controllers;
+
+[Authorize]
+[ApiController]
+[Route("api/claim-automation")]
+public class ClaimController : ControllerBase
+{
+    private readonly IClaimRepository claimRepository;
+
+    public ClaimController(IClaimRepository claimRepository)
+    {
+        this.claimRepository = claimRepository;
+    }
+
+    [HttpPost("start")]
+    [ProducesResponseType(typeof(ClaimSubject), 201)]
+    [ProducesResponseType(typeof(ClaimSubject), 200)]
+    public async Task<ActionResult> StartClaimProcess()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var claim = await claimRepository.GetClaimSubject(Guid.Parse(subject));
+
+        if (claim != null)
+        {
+            return Ok(claim);
+        }
+
+        var claimSubject = new ClaimSubject(Guid.Parse(subject));
+
+        try
+        {
+            claim = await claimRepository.AddClaimSubject(claimSubject);
+            return CreatedAtAction(nameof(GetClaimProcess), null, claim);
+        }
+        catch (DbUpdateException)
+        {
+            return CreatedAtAction(nameof(GetClaimProcess), null, claimSubject);
+        }
+    }
+
+    [HttpDelete("stop")]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(404)]
+    public async Task<ActionResult> StopClaimProcess()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var claim = await claimRepository.GetClaimSubject(Guid.Parse(subject));
+        if (claim == null)
+        {
+            return NotFound();
+        }
+
+        claimRepository.DeleteClaimSubject(claim);
+        return NoContent();
+    }
+
+    [HttpGet("history")]
+    [ProducesResponseType(404)]
+    [ProducesResponseType(typeof(ClaimSubjectHistoryEntriesDto), 200)]
+    public async Task<ActionResult<ClaimSubjectHistoryEntriesDto>> GetClaimProcessHistory()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var history = await claimRepository.GetHistory(Guid.Parse(subject));
+
+        if (history.Count == 0)
+        {
+            return NotFound();
+        }
+
+        var historyDto = history.Select(c =>
+                new ClaimSubjectHistoryEntryDto
+                {
+                    ActorName = c.ActorName,
+                    CreatedAt = c.CreatedAt,
+                    AuditAction = c.AuditAction
+                }
+            ).ToList();
+
+        return Ok(new ClaimSubjectHistoryEntriesDto(historyDto));
+    }
+
+    [HttpGet]
+    [ProducesResponseType(404)]
+    [ProducesResponseType(200)]
+    public async Task<ActionResult<ClaimSubject>> GetClaimProcess()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var claim = await claimRepository.GetClaimSubject(Guid.Parse(subject));
+        if (claim == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(claim);
```

Comment:
Have you considered returning when the claim automation was started?

Reply:
No i havent. The frontend is not using the timestamp

Reply:
This is returning an object of type `ClaimSubject`, which only has one property and that is the subject id. Technically, the frontend does not even need the subject id returned. 
But I just thought it would make sense to at least store a timestamp for when the claim automation was started.

Reply:
Added CreatedAt to ClaimSubject so it now returns timestamp as well